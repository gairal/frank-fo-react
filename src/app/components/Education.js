import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Grid } from '@material-ui/core';
import XpCard from './core/XpCard';

export default class Education extends Component {
  static propTypes = {
    load: PropTypes.func.isRequired,
    educations: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  };

  componentDidMount() {
    const { load } = this.props;
    load();
  }

  render() {
    const { educations } = this.props;
    return (
      <section>
        <Grid container>
          {educations.map(edu => (
            <Grid item xs={12} key={edu.year_in}>
              <XpCard
                img={edu.image.name}
                name={edu.name}
                title={edu.short_description}
                date_in={edu.year_in}
                date_out={edu.year_out}
                place={edu.place}
                subHeadline={edu.diploma}
              />
            </Grid>
          ))}
        </Grid>
      </section>
    );
  }
}
