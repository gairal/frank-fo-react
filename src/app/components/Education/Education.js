import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { CircularProgress } from 'material-ui/Progress';
import { Grid } from 'material-ui';


class Education extends Component {
  static propTypes = {
    load: PropTypes.func.isRequired,
    educations: PropTypes.arrayOf(PropTypes.object),
    isFetching: PropTypes.bool,
  }

  static defaultProps = {
    educations: [],
    isFetching: false,
  }

  componentDidMount() {
    this.props.load();
  }

  render() {
    return (
      <Grid item xs={12}>
        <Grid container>
          {(this.props.isFetching
            ? <CircularProgress />
            : null
          )}
          {this.props.educations.map(edu => (
            <Grid
              item
              xs={3}
              key={edu.id}
            >
              Education
            </Grid>
          ))}
        </Grid>
      </Grid>
    );
  }
}

export default Education;
