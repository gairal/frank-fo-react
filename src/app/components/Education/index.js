import education from './Education';

export default education;
