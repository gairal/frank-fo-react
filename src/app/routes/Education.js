import component from '@/components/Education';
import AbstractRoute from './AbstractRoute';

class Education extends AbstractRoute {
  constructor(store) {
    super(store, 'educations', 'account_balance');
    this.component = component;

    this.ACTIONS = {
      FETCH_SUCCESS: 'EDU_FETCH_SUCCESS',
      FETCH_FAILURE: 'EDU_FETCH_FAILURE',
    };
    this.init();
  }

  // actions
  static success(json) {
    return {
      type: 'EDU_FETCH_SUCCESS',
      payload: {
        json,
      },
    };
  }

  static fail(err) {
    return {
      type: 'EDU_FETCH_FAILURE',
      payload: {
        err,
      },
    };
  }
}

export default store => new Education(store).route;
