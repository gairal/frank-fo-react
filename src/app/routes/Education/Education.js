import AbstractRoute from '../AbstractRoute';
import component from '../../components/Education';

export default class Work extends AbstractRoute {
  constructor(store) {
    super(store, 'educations', 'account_balance');
    this.component = component;

    this.ACTIONS = {
      FETCH_SUCCESS: 'EDU_FETCH_SUCCESS',
      FETCH_FAILURE: 'EDU_FETCH_FAILURE',
    };
    this.init();
  }

  // actions
  static success(json) {
    return {
      type: 'EDU_FETCH_SUCCESS',
      payload: {
        json,
      },
    };
  }

  static fail(err) {
    return {
      type: 'EDU_FETCH_FAILURE',
      payload: {
        err,
      },
    };
  }
}
