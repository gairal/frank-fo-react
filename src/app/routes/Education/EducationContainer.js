import { connect } from 'react-redux';
import { add, remove } from '../Education';

import Experience from '../../components/Education';

const mapDispatchToProps = {
  add,
  remove,
};

const mapStateToProps = state => ({
  todos: state.todo,
});

export default connect(mapStateToProps, mapDispatchToProps)(Experience);
