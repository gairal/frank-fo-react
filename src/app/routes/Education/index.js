import Education from './Education';

export default store => new Education(store).route;
