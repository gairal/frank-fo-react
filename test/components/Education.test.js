import React from 'react';
import renderer from 'react-test-renderer';
import Education from '@/components/Education';

describe('components > Education', () => {
  it('Renders correctly default', () => {
    const component = renderer.create(
      <Education
        load={jest.fn}
        educations={[
          {
            name: 'test',
            diploma: 'test',
            place: 'test',
            year_in: '2018-01-01',
            description: 'test',
            short_description: 'test',
            image: { name: 'test' },
          },
        ]}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
