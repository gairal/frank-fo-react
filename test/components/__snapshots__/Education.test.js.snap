// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > Education Renders correctly default 1`] = `
<section>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-40"
    >
      <div
        className="MuiPaper-root-98 MuiPaper-elevation4-104 MuiPaper-rounded-99 MuiCard-root-97"
        style={
          Object {
            "display": "flex",
            "marginBottom": "1rem",
            "padding": "1rem",
          }
        }
      >
        <div
          className="MuiCardMedia-root-125 card-media"
          style={
            Object {
              "backgroundImage": "url(\\"/img/test\\")",
              "flex": "none",
            }
          }
          title="test"
        />
        <div
          className="MuiCardContent-root-127"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <h2
            className="MuiTypography-root-128 MuiTypography-title-134"
          >
            test
             | 
            test
          </h2>
          <div
            className="MuiGrid-container-1 card-chips"
            style={
              Object {
                "textTransform": "uppercase",
              }
            }
          >
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-md-true-56"
              style={
                Object {
                  "flexGrow": 0,
                  "marginBottom": ".5rem",
                  "marginRight": "0.2rem",
                }
              }
            >
              <div
                className="MuiChip-root-152"
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <div
                  className="MuiAvatar-root-159 MuiAvatar-colorDefault-160 MuiChip-avatar-155"
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-162 MuiChip-avatarChildren-156"
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": "19px",
                      }
                    }
                  >
                    date_range
                  </span>
                </div>
                <span
                  className="MuiChip-label-157"
                >
                  2017 December - present
                </span>
              </div>
            </div>
            <div
              className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-md-true-56"
            >
              <div
                className="MuiChip-root-152"
                onKeyDown={[Function]}
                role="button"
                tabIndex={-1}
              >
                <div
                  className="MuiAvatar-root-159 MuiAvatar-colorDefault-160 MuiChip-avatar-155"
                >
                  <span
                    aria-hidden="true"
                    className="material-icons MuiIcon-root-162 MuiChip-avatarChildren-156"
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": "19px",
                      }
                    }
                  >
                    location_city
                  </span>
                </div>
                <span
                  className="MuiChip-label-157"
                >
                  test
                </span>
              </div>
            </div>
          </div>
          <p
            className="MuiTypography-root-128 MuiTypography-caption-138"
            style={
              Object {
                "marginTop": "0.5rem",
              }
            }
          >
            test
          </p>
        </div>
      </div>
    </div>
  </div>
</section>
`;
